actions/produce.py

```diff
@@ -28,18 +33,28 @@ def run(self, topic, message, hosts=None):
 
         if hosts:
             _hosts = hosts
-        elif self.config.get('hosts', None):
-            _hosts = self.config['hosts']
+        elif self.config.get("hosts", None):
+            _hosts = self.config["hosts"]
         else:
             raise ValueError("Need to define 'hosts' in either action or in config")
 
         # set default for empty value
-        _client_id = self.config.get('client_id') or self.DEFAULT_CLIENT_ID
-
-        client = KafkaClient(_hosts, client_id=_client_id)
-        client.ensure_topic_exists(topic)
-        producer = SimpleProducer(client)
-        result = producer.send_messages(topic, kafka_bytestring(message))
-
-        if result[0]:
-            return result[0]._asdict()
+        _client_id = self.config.get("client_id") or self.DEFAULT_CLIENT_ID
+
+        consumer = KafkaConsumer(
+            bootstrap_servers=_hosts.split(","), client_id=_client_id
+        )
+        try:
+            assert topic in consumer.topics()
+        except AssertionError as exc:
+            raise AssertionError(f"{topic} does not exist.") from exc
+
+        producer = KafkaProducer(
+            bootstrap_servers=_hosts.split(","),
+            client_id=_client_id,
+            value_serializer=lambda m: m.encode("utf-8"),
+            max_request_size=10485760,
+        )
+        result = producer.send(topic, message)
+        record_metadata = result.get(timeout=10)
+        return record_metadata
```

Comment:
Please return a dict instead of a RecordMetadata (a namedtuple that gets dumped as a json list).

Also, I think the timeout should be configurable, but that can wait for a follow-up.

```suggestion
        future = producer.send(topic, message)
        record_metadata = future.get(timeout=10)  # TODO: make this timeout an input param
        return record_metadata._asdict()
```